src/CSI.IBTA.BenefitsService/Services/InsurancePackageService.cs

```diff
@@ -104,39 +99,30 @@ public async Task<GenericResponse<List<InsurancePackageDto>>> GetInsurancePackag
 
         public async Task<GenericResponse<FullInsurancePackageDto>> GetInsurancePackage(int packageId)
         {
-            try
-            {
-                var plans = await _benefitsUnitOfWork.Plans.Include(x => x.PlanType)
-                    .Where(x => x.PackageId == packageId).ToListAsync();
+            var plans = await _benefitsUnitOfWork.Plans.Include(x => x.PlanType)
+                .Where(x => x.PackageId == packageId).ToListAsync();
 
-                var package = await _benefitsUnitOfWork.Packages.GetById(packageId);
-
-                if (package == null)
-                {
-                    var error = new HttpError("Insurance package not found", HttpStatusCode.NotFound);
-                    return new GenericResponse<FullInsurancePackageDto>(error, null);
-                }
+            var package = await _benefitsUnitOfWork.Packages.GetById(packageId);
```

Comment:
plans and package could be awaited in a parallel as
Task.WhenAll(plansTask, packageTask). 

It is a minor perf update, so it is OK keep the code as is